src/EndpointManager.sol

```diff
@@ -207,9 +208,8 @@ abstract contract EndpointManager is IEndpointManager, OwnableUpgradeable, Reent
         pure
         returns (bytes memory encoded)
     {
-        // TODO -- should we check payload length here?
-        // for example, CCTP integration checks payload is <= max(uint16)
-        return abi.encodePacked(m.chainId, m.sequence, m.msgType, m.payload);
+        uint16 payloadLength = uint16(m.payload.length);
+        return abi.encodePacked(m.chainId, m.sequence, m.msgType, payloadLength, m.payload);
```

Comment:
Should we validate the payload length is `<= max(uint16)`, similar to the CCTP contracts?

Reply:
yes good point

Reply:
done